app/products/calls/connection/connection.ts

```diff
@@ -68,15 +74,48 @@ export async function newConnection(
         }
 
         try {
-            stream = await mediaDevices.getUserMedia({
+            voiceStream = await mediaDevices.getUserMedia({
                 video: false,
                 audio: true,
             }) as MediaStream;
-            voiceTrack = stream.getAudioTracks()[0];
+            voiceTrack = voiceStream.getAudioTracks()[0];
             voiceTrack.enabled = false;
-            streams.push(stream);
+            streams.push(voiceStream);
         } catch (err) {
-            logError('calls: unable to get media device:', err);
+            logError('calls: unable to get audio stream:', err);
+        }
+    };
+
+    const initializeVideoTrack = async (start: boolean) => {
+        if (videoTrack) {
+            return;
+        }
+
+        try {
+            videoStream = await mediaDevices.getUserMedia({
+                video: {
+                    frameRate: 30,
+                    facingMode: 'user', // TODO: allow user to switch camera
+                },
+                audio: false,
+            }) as MediaStream;
+            videoTrack = videoStream.getVideoTracks()[0];
+
+            logDebug('calls: video track initialized', videoTrack.id, videoStream.id);
+
+            if (start) {
+                streams.push(videoStream);
+                setLocalVideoURL(videoStream.toURL());
+                videoTrack.enabled = true;
+            } else {
+                // If we're not starting the video, we release any resources immediately.
```

Comment:
There's a chance we don't need to do this ever. I've kept this in line with the web side where this case is essentially used to ensure the camera works but since here permission checks are decoupled perhaps the extra logic is not needed.